src/storage/AccountStorageProvider.tsx

```diff
@@ -48,6 +52,19 @@ const useAccountStorageHook = () => {
   const solanaAccountsUpdateComplete = useRef(false)
   const solanaContactsUpdateComplete = useRef(false)
   const { updateL1Network, l1Network } = useAppStorage()
+  const dispatch = useAppDispatch()
+  const currentAppState = useAppState()
+
+  const updateApiToken = useCallback(async () => {
+    const apiToken = await makeApiToken(currentAccount?.address)
+    dispatch(authSlice.actions.setApiToken(apiToken))
+  }, [currentAccount, dispatch])
+
+  useEffect(() => {
+    if (currentAppState === 'active') {
```

Comment:
So this will be called every time the app state changed back to active? Should we add a condition to check if you already have an api token set? Or do we want it to update every time? Like if you background and foreground it would probably get called again?

Reply:
Yeah. The idea is that it is expiring. So if someone were to intercept your API token, it will expire within an hour.

Reply:
The time is used to enforce on the backend